core/domain/collection_domain.py

```diff
@@ -345,6 +346,72 @@ def from_dict(
 
         return collection
 
+    @classmethod
+    def deserialize(cls, json_string):
+        """Return a Collection domain object decoded from a JSON string.
+
+        Args:
+            json_string: str. A JSON-encoded string that can be
+                decoded into a dictionary representing a Collection. Only call
+                on strings returned from the platform.cache cache_services.
+
+        Returns:
+            Collection. The corresponding Collection domain object.
+        """
+        collection_dict = json.loads(json_string)
+
+        created_on = (
+            utils.convert_string_to_naive_datetime_object(
+                collection_dict['created_on'])
+            if 'created_on' in collection_dict else None)
+        last_updated = (
+            utils.convert_string_to_naive_datetime_object(
+                collection_dict['last_updated'])
+            if 'last_updated' in collection_dict else None)
+        collection = cls.from_dict(
+            collection_dict,
+            collection_version=collection_dict['version'],
+            collection_created_on=created_on,
+            collection_last_updated=last_updated)
+
+        return collection
+
+    def serialize(self):
+        """Returns the object serialized as a JSON string.
+
+        Returns:
+            str. JSON-encoded string encoding all of the information composing
+            the object.
+        """
+        collection_dict = {
+            'id': self.id,
+            'title': self.title,
+            'category': self.category,
+            'objective': self.objective,
+            'language_code': self.language_code,
+            'tags': self.tags,
+            'schema_version': self.schema_version,
+            'nodes': [
+                node.to_dict() for node in self.nodes
+            ],
+            'version': self.version,
+            'created_on': utils.convert_naive_datetime_to_string(
+                self.created_on),
+            'last_updated': utils.convert_naive_datetime_to_string(
+                self.last_updated)
+        }
+
+        if self.created_on:
+            collection_dict['created_on'] = (
+                utils.convert_naive_datetime_to_string(self.created_on))
+
+        if self.last_updated:
+            collection_dict['last_updated'] = (
+                utils.convert_naive_datetime_to_string(self.last_updated))
+
+        return json.dumps(collection_dict)
+
+
```

Comment:
Drop one newline -- adjacent methods in a class should be separated by exactly one newline.